lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -204,7 +204,8 @@ sub login {
     _init_db($request);
     sanity_checks($database);
     $request->{login_name} = $version_info->{username};
-    if ($version_info->{status} eq 'does not exist'){
+    # $version_info->{status} isn't always defined
```

Comment:
The reason you're checking "$version_info->{status}" for defined-ness on this level is obvious. A better comment explains why it's not defined. Can you add a comment to that effect?